CMakeLists.txt

```
@@ -743,6 +743,13 @@ endif (WITH_PKGCONFIG)
 
 # Unit testing
 
+if (BUILD_FUZZING)
+  add_executable (fuzz_demangle
+    src/fuzz_demangle.cc
+  )
+  target_link_libraries(fuzz_demangle glog $ENV{LIB_FUZZING_ENGINE})
+endif (BUILD_FUZZING)
```

I would use a CMake variable here. The caller can still use an environment variable if he chooses to do so.
```suggestion
if (BUILD_FUZZING AND WITH_FUZZING_ENGINE)
  add_executable (fuzz_demangle
    src/fuzz_demangle.cc
  )
  target_link_libraries (fuzz_demangle PRIVATE glog ${WITH_FUZZING_ENGINE})
endif (BUILD_FUZZING AND WITH_FUZZING_ENGINE)
```
```suggestion
  target_link_libraries (fuzz_demangle PRIVATE glog ${WITH_FUZZING_ENGINE})
```
`WITH_FUZZING_ENGINE` should be populated with sensible choices including one that disables fuzzing altogether (e.g., `NO` could work) which should also probably be the default.

---

Good idea for the cmake variable `WITH_FUZZING_ENGINE`.

I am not sure I understand what you expect by `should be populated with sensible choices` (maybe I do not know cmake enough)

---

See my additional comments.